src/osdc/Objecter.cc

```diff
@@ -2884,10 +2884,12 @@ int Objecter::_calc_target(op_target_t *t, Connection *con, bool any_change)
     force_resend = true;
   }
 
-  bool unpaused = false;
-  if (t->paused && !target_should_be_paused(t)) {
+  bool unpaused = !target_should_be_paused(t);
+  if (t->paused && unpaused) {
     t->paused = false;
```

Comment:
if it gets here, unpaused is true, so false == !unpaused

looks like this whole block would be replaced by 

`t->paused = target_should_be_paused(t);`

the else statement is hit either if
* `t` is not currently paused, in which case we pause it if it should be `!unpaused`
* `t` is paused and it should be, then we keep it that way... `true = true`

thus the var and the condition can be eliminated by simply...

`t->paused = target_should_be_paused(t);`

am I missing something ? of course unit tests would clarify this point

Reply:
We actually use ```unpaused``` way farther down the function (https://github.com/ceph/ceph/pull/25243/files#diff-847b3447bebb2f031678eef0a2aea60dR2971) and this whole dance is about getting both that and ```t->paused``` correct.

Reply:
@runsisi Can you do that in a follow-up pr? That makes the code much simpler and hence readable.

Thanks!